invenio_rdm_records/services/services.py

```diff
@@ -198,44 +199,41 @@ def discard_pid(self, id_, identity, pid_type, pid_client=None):
             links_tpl=self.links_item_tpl,
         )
 
-    def _lift_embargo_from(self, rdm_record):
-        """Lifts embargo from record or draft."""
-        if not rdm_record.access.embargo.lift():
-            raise EmbargoNotLiftedError(rdm_record["id"])
-        rdm_record.access.protection.record = "public"
-        rdm_record.access.protection.files = "public"
-
     def lift_embargo(self, _id, identity):
-        """Lifts embargo from the record and updates draft."""
-
-        def draft_access_field_was_modified(draft, record):
-            """Returns True if draft's access field was modified."""
-            return draft.get('access') == record.get('access')
+        """Lifts embargo from the record and draft (if exists).
 
+        It's an error if you try to lift an embargo that has not yet expired.
+        Use this method in combination with scan_expired_embargos().
+        """
         # Get the record
         record = self.record_cls.pid.resolve(_id)
 
         # Check permissions
         self.require_permission(identity, "lift_embargo", record=record)
 
-        lifted_embargo_from_draft = False
-        # Check if record has already a draft
+        # Modify draft embargo if draft exists and it's the same as the record.
+        draft = None
         if record.has_draft:
             draft = self.draft_cls.pid.resolve(_id, registered_only=False)
-            # If the draft has no modifications in the access field the
-            # embargo is lifted
-            if draft_access_field_was_modified(draft, record):
-                # Lifts embargo from draft
-                self._lift_embargo_from(draft)
-                lifted_embargo_from_draft = True
-
-        # Lifts embargo from record
-        self._lift_embargo_from(record)
-        # Commit and index
+            if record.access == draft.access:
+                draft.access.lift_embargo()
+                draft.commit()
+
+        if not record.access.lift_embargo():
```

Comment:
The embargo from the draft is being lifted before, and it's result is ignored (and commited). Since the access conditions are checked right before. Would it be sensible to raise also from the embargo lift of the draft (line 219 above)? The previous implementation was raising on both cases.

It might save a few DB queries and the result would be the same? Alternatively, lift first the record and then the draft.

Reply:
Yep, i'll move the record lift up first.